chirpstack/src/api/tenant.rs

```diff
@@ -158,11 +158,16 @@ impl TenantService for Tenant {
         let auth_id = request.extensions().get::<AuthID>().unwrap();
         let req = request.get_ref();
         let mut filters = tenant::Filters::default();
+        let user_id = Uuid::from_str(&req.user_id).map_err(|e| e.status())?;
 
         if !req.search.is_empty() {
             filters.search = Some(req.search.clone());
         }
 
+        if !req.user_id.is_empty() {
+            filters.user_id = Some(user_id);
```

Comment:
This must be placed within the `AuthID::Key(_) => {` match arm, as this filter can only be set for global API keys.

Please note that you also need to implement `filters.user_id`. See: https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/storage/tenant.rs#L95

You must also implement this filter here:
https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/storage/tenant.rs#L175
https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/storage/tenant.rs#L201


Then also please add a test for this filter here:
https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/storage/tenant.rs#L432

Reply:
> This must be placed within the AuthID::Key(_) => { match arm, as this filter can only be set for global API keys.
It would have gotten replaced by the clause for non-admin where it replaces the user_id for the id of the user making the request, but this is cleaner so I have pushed it.

In terms of the rest of the changes, `user_id` is already a filter listed (currently used for when it is non-admin and the list needs to be filtered for the tenants of the user making the request). Would it still need to be added as a separate filter when using it as an admin? I'm new on rust so I'm trying to understand it but not sure what the difference would be.

Reply:
@brocaar still having trouble understanding what needs to change in 
https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/storage/tenant.rs#L95
https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/storage/tenant.rs#L175
https://github.com/chirpstack/chirpstack/blob/master/chirpstack/src/storage/tenant.rs#L201

since it already uses `user_id` filter for other instances (when the request is from non-admin), shouldn't it work the same? I must be missing something. Appreciate the help!

Reply:
Ah yes, you are right, the `user_id` filter is already there.